src/main/java/com/hierynomus/smbj/share/Share.java

```diff
@@ -42,6 +44,11 @@
 import java.util.concurrent.atomic.AtomicBoolean;
 
 public class Share implements AutoCloseable {
+    private static final SMB2FileId ROOT_ID = new SMB2FileId(
```

Comment:
I considered that option. I don't think this is used anywhere except for operations that are targeted at the share itself. That's why it's located in Share. Making it a public constant on SMB2FileId seems to communicate that it's generally useful which I don't think it is.
No objections to moving it though if you prefer to have a private constructor. I don't think there's much benefit in making the constructor private though. You can always still create an instance using the static factory method.